lib/collection/src/collection.rs

```diff
@@ -258,17 +258,79 @@ impl Collection {
         todo!("Handle for sender and receiver")
     }
 
+    pub async fn initiate_temporary_shard(&self, shard_id: ShardId) -> CollectionResult<()> {
+        let shard_holder_read = self.shards_holder.read().await;
+        // validate that the shard is known as a remote shard
+        if let Some(Shard::Remote(_)) = shard_holder_read.get_shard(&shard_id) {
```

Comment:
Same here for support of non-existing remote shard. I suppose we need to do something like:

- Check if temp shard exists
  - if exists - drop it and re-create the directory
- Crete a new temporary shard


Do not check if remote shard exists, as it is not a strict requirement.

Reply:
I removed the remote shard requirement & added a bit of cleanup for existing temporary shards.
https://github.com/qdrant/qdrant/pull/829/commits/46f5b4b0d337d8b2c455936a1940c61b13bd6cc1